packages/vscode-container-client/src/clients/DockerComposeClientBase/DockerComposeClientBase.ts

```diff
@@ -0,0 +1,254 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { GeneratorCommandResponse, PromiseCommandResponse, VoidCommandResponse } from '../../contracts/CommandRunner';
+import {
+    CommonOrchestratorCommandOptions,
+    ConfigCommandOptions,
+    ConfigItem,
+    DownCommandOptions,
+    IContainerOrchestratorClient,
+    LogsCommandOptions,
+    RestartCommandOptions,
+    StartCommandOptions,
+    StopCommandOptions,
+    UpCommandOptions,
+} from '../../contracts/ContainerOrchestratorClient';
+import {
+    CommandLineArgs,
+    CommandLineCurryFn,
+    composeArgs,
+    withArg,
+    withFlagArg,
+    withNamedArg,
+    withVerbatimArg,
+} from '../../utils/commandLineBuilder';
+import { stringStreamToGenerator } from '../../utils/streamToGenerator';
+import { ConfigurableClient } from '../ConfigurableClient';
+
+function withCommonOrchestratorArgs(options: CommonOrchestratorCommandOptions): CommandLineCurryFn {
+    return composeArgs(
+        withNamedArg('--file', options.files),
+        withNamedArg('--env-file', options.environmentFile),
+        withNamedArg('--project-name', options.projectName),
+        withNamedArg('--profile', options.profiles),
+    );
+}
+
+function withComposeArg(composeV2: boolean): CommandLineCurryFn {
+    // If using Compose V2, then add the `compose` argument at the beginning
+    // That way, the command is `docker compose` instead of `docker-compose`
+    return withArg(composeV2 ? 'compose' : undefined);
+}
+
+export abstract class DockerComposeClientBase extends ConfigurableClient implements IContainerOrchestratorClient {
+
+    /**
+     * Whether to use Docker Compose V2 or not.
+     * If true, the command will be like `docker compose` instead of `docker-compose`.
+     */
+    public composeV2: boolean = true;

```

Comment:
This is just a public field now instead of a public getter/setter + private field.